### PR TITLE
Reduse the use of e_os.h in test programs

### PR DESCRIPTION
**Pull request by levitte**
This includes unnecessary use of the top as inclusion directory

### CODE REVIEW

**Review by mattcaswell — APPROVED** [review]

**Comment by bernd-edlinger** [comment]
travis & appveyor build failures look relevant.
BTW: In the commit message, "Reduce" instead of "Reduse" ?

**Comment by levitte** [comment]
Commit message fixed.
Also fixed the CI issues (I hope)

File: test/bntest.c [hunk 1]
```diff
@@ -6,35 +6,23 @@
...
- *
- * To avoid this, we define a dummy variant of bn_expand2() here, and to
- * avoid possible clashes with libcrypto, we rename it first, using a macro.
- */
-#define bn_expand2 dummy_bn_expand2
-BIGNUM *bn_expand2(BIGNUM *b, int words);
-BIGNUM *bn_expand2(BIGNUM *b, int words) { return NULL; }
-#include "../crypto/bn/bn_lcl.h"
+#ifdef OPENSSL_SYS_WINDOWS
+# defined strcasecmp _stricmp
```
Comment (mattcaswell) [inline test/bntest.c]: define?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: test/bntest.c (hunk 1 above)
Comment (levitte) [reply test/bntest.c]: Fixed.

Time for a pot of coffee, methinks...

File: test/bntest.c [hunk 2]
```diff
@@ -6,35 +6,23 @@
...
 #include <ctype.h>
 
-#include "internal/nelem.h"
-#include "internal/numbers.h"
 #include <openssl/bn.h>
 #include <openssl/crypto.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
+#include "internal/nelem.h"
+#include "internal/numbers.h"
```
Comment (levitte) [inline test/bntest.c]: Should anyone wonder, `stdint.h` is included by at least one of those `<openssl/{whatever}.h>`, and that needs to happen before the inclusion of `"internal/numbers.h"`.  Either way, it makes sense to include out internal headers after the public/system ones, as the former are designed to fix up things that weren't defined by the latter...

File: test/ssl_test_ctx.c [hunk 3]
```diff
@@ -7,7 +7,6 @@
  * https://www.openssl.org/source/license.html
  */
 
-#include "../e_os.h"
```
Comment (bernd-edlinger) [inline test/ssl_test_ctx.c]: another strcasecmp down there...

Comment (levitte) [reply test/ssl_test_ctx.c]: Yup, noticed it.

**Comment by levitte** [comment]
It seems Travis is fine at this point, all that's needed is to battle it out with Appveyor...

**Comment by levitte** [comment]
The CIs look happy now

**Review by bernd-edlinger — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged. 2b1aa1988189773497d6edba443cf77f5c31feba

Note that this was master only.  Would it be valuable to do this for 1.1.0 as well?  I didn't cherry-pick, 'cause I'm sure there are too many differences to make it cleanly

**Comment by bernd-edlinger** [comment]
1.1.0 has many more of these, but I don't care.

While you are already at it, have you seen these?

```
test/ossl_shim/packeted_bio.h:#include "../../e_os.h"
test/ssltest_old.c:#include "e_os.h"
```

**Comment by levitte** [comment]
I'm ignoring `test/ssltest_old.c`, as it's mainly replaced by `test/ssl_test.c`.  There are only two cases left with this old test program...

**Comment by richsalz** [comment]
It's a cleanliness issue, not a bugfix.  Don't bother backporting.

**Comment by levitte** [comment]
Okie.

**Pull request closed without merging**